### PR TITLE
fix: reduce flakiness by refactoring unit test error/proto comparisons

### PR DESCRIPTION
**Pull request by obreitwi**
* aim to reduce flakiness of tests due to protojson instabilities

Rev: SRX-UG7OJV

### CODE REVIEW

File: README_DEVELOPERS.md [hunk 1]
```diff
@@ -120,6 +120,59 @@ cd services/cd-service
 go test ./... -v
 ```
 
+### Best practices for unit tests
+
+#### Always use `cmp.Diff`
+
+When writing unit tests, aim to always compare with `cmp.Diff` and print the result.
+For errors, the the test should check via:
```
Comment (sven-urbanski-freiheit-com) [inline README_DEVELOPERS.md:128]:
```suggestion
For errors, the test should check via:
```

File: pkg/auth/azure_test.go [hunk 2]
```diff
@@ -28,40 +28,54 @@ import (
...
+	msg string
+}
+
+func (e errMatcher) Error() string {
+	return e.msg
+}
+
+func (e errMatcher) Is(err error) bool {
+	return e.Error() == err.Error()
+}
```
Comment (sven-urbanski-freiheit-com) [inline pkg/auth/azure_test.go:45]: this code seems quite duplicated. Can we move it into testutil.go?

**Review by sven-urbanski-freiheit-com — CHANGES_REQUESTED** [review]
(see other comments)

File: pkg/auth/azure_test.go (hunk 2 above)
Comment (obreitwi) [reply pkg/auth/azure_test.go:45]: Was not sure if I should introduce coupling between tests, but can move the code, yes.

File: services/cd-service/pkg/repository/transformer_test.go [hunk 3]
```diff
@@ -5553,12 +5586,18 @@ func TestAllErrorsHandledDeleteEnvironmentLock(t *testing.T) {
 func TestAllErrorsHandledDeleteEnvironmentApplicationLock(t *testing.T) {
 	t.Parallel()
 	collector := &testfs.UsageCollector{}
+	fixtureWrapTransformError := func(err error) *TransformerBatchApplyError {
```
[the remark this answers is not in the document]
Comment (obreitwi) [reply services/cd-service/pkg/repository/transformer_test.go:5589]: Was not sure if I should already add them there, keeping them local to the test makes them easier to change, but can move to `testutil.go`.

**Review by sven-urbanski-freiheit-com — APPROVED** [review]

File: services/cd-service/pkg/repository/transformer_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (sven-urbanski-freiheit-com) [reply services/cd-service/pkg/repository/transformer_test.go:5589]: let's keep it as it is, to get this merged quickly

**Pull request merged**